### PR TITLE
Implement design feedback and fix tests

### PR DESCRIPTION
**Pull request by romeovs**
### Description

- Implements some fixes from [Maz's feedback](https://metaboat.slack.com/archives/C0645JP1W81/p1708577874938219?thread_ts=1706924051.845229&cid=C0645JP1W81).
- Fixes a flaky test case for mongo

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx [hunk 1]
```diff
@@ -6,8 +6,8 @@ import type * as Lib from "metabase-lib";
 
 import { FilterColumnName } from "../FilterColumnName";
 
-import { InfoIcon, HoverParent } from "./FilterTitle.styled";
-export { HoverParent };
+import { InfoIcon } from "./FilterTitle.styled";
+export { HoverParent } from "metabase/components/MetadataInfo/ColumnInfoIcon";
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx:10]: This violates `import/order` rule :thinking:

**Review by kamilmielnik — APPROVED** [review]

File: frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx (hunk 1 above)
Comment (romeovs) [reply frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx:10]: Strange, I would've expected it to come up in the linter.

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx]: Oh, it's an export, I didn't see that before.

Why are we exporting it from here?

Comment (romeovs) [reply frontend/src/metabase/querying/components/FilterModal/FilterTitle/FilterTitle.tsx]: It's coupled to the `InfoTitle` component (in the file it's being exported from), so I feel like the exports should be kept together.

**Pull request merged**